src/wallet/wallet.cpp

```diff
@@ -4478,10 +4478,15 @@ void CWallet::LoadActiveScriptPubKeyMan(uint256 id, OutputType type, bool intern
 {
     WalletLogPrintf("Setting spkMan to active: id = %s, type = %d, internal = %d\n", id.ToString(), static_cast<int>(type), static_cast<int>(internal));
     auto& spk_mans = internal ? m_internal_spk_managers : m_external_spk_managers;
+    auto& spk_mans_other = internal ? m_external_spk_managers : m_internal_spk_managers;
     auto spk_man = m_spk_managers.at(id).get();
     spk_man->SetInternal(internal);
     spk_mans[type] = spk_man;
 
+    if (spk_mans_other[type] == spk_man) {
+        spk_mans_other[type] = nullptr;
```

Comment:
I analyzed the code and couldn't find any uses of `LoadActiveScriptPubKeyMan` for legacy wallets. The function is only called when 1) we create new wallet, 2) call `importdescriptors` command or 3) when loading a wallet from disk. But if I'm not mistaken there is no active SPKs on disk for legacy wallets. That makes me believe this function is only relevant for descriptor wallets. So I put a guard at the top to throw an exception when we call it for legacy wallet. And all wallet functional tests pass on my laptop for now. I'm happy to reproduce any scenario in which this code should break legacy wallet.

Reply:
Hmm, `SetupLegacyScriptPubKeyMan` adds the `LegacySPKM` to the maps that track internal/external, but doesn't write them to disk. I thought it wrote them to disk and now I'm wondering whether not doing so is a bug. But since it doesn't, that's why this works.

Reply:
cead7ed ISTM this change should have test coverage (more so given the discussion) to spec the behavior and provide regression coverage. Edit: Ah, if I understand correctly some test coverage is added in the next commit b66b26a.